python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -705,13 +721,56 @@ def get_lsst_magnitudes(self):
         """
         getter for LSST magnitudes of solar system objects
         """
-
         if not hasattr(self, 'lsstBandpassDict'):
             self.lsstBandpassDict = BandpassDict.loadTotalBandpassesFromFiles()
 
         return self._magnitudeGetter(self.lsstBandpassDict, self.get_lsst_magnitudes._colnames)
 
 
+    def get_magFilter(self):
+        """
+        Generate the magnitude in the filter of the observation.
+        """
```

Comment:
Can you not just do:

magName = 'lsst_' + self.obs_metadata.bandpass
return self.column_by_name(magName)

Then, all of the work of loading the bandpassDict will be done by the compound getter for lsst_[u,g,r,i,z,y].